src/main/java/nl/knaw/dans/filemigration/core/EasyFileLoader.java

```diff
@@ -0,0 +1,139 @@
+
+  void fedoraFiles(FedoraToBagCsv csv) {
+    log.trace(csv.toString());
+    // read fedora files before adding expected migration files
+    // thus we don't write anything when reading fails
+    List<EasyFile> easyFiles = getByDatasetId(csv);
+    for (EasyFile f: easyFiles) {
+      // note: biggest pdf/image option for europeana in easy-fedora-to-bag does not apply to migration
+      ExpectedFile expected = transformedFedoraFile(csv, f);
+      try {
+        saveExpected(expected);
+      } catch(PersistenceException e){
+        // logged as error by org.hibernate.engine.jdbc.spi.SqlExceptionHelper
+        if (!(e.getCause() instanceof ConstraintViolationException))
+          throw e;
+        else {
+          if (expected.getRemoved_duplicate_file_count() > 10) {
+            // TODO temporary safe guard?
+            log.error("too many retries on duplicate file, skipping: {}", expected);
+          }
+          else {
+            expected.incRemoved_duplicate_file_count();
+            saveExpected(expected);
+          }
+        }
+      }
+    }
+  }
+
+  public List<EasyFile> getByDatasetId(FedoraToBagCsv csv) {
+    return easyFileDAO.findByDatasetId(csv.getDatasetId());
+  }
+
+  public void saveExpected(ExpectedFile expected) {
+      expectedDAO.create(expected);
+  }
+
+  private static ExpectedFile addedMigrationFile(FedoraToBagCsv csv, String migrationFile) {
+    ExpectedFile expected = new ExpectedFile();
+    expected.setDoi(csv.getDoi());
+    expected.setSha1_checksum("");
+    expected.setEasy_file_id("");
+    expected.setFs_rdb_path("");
+    expected.setExpected_path("migration/" + migrationFile);
+    expected.setAdded_during_migration(true);
+    expected.setRemoved_thumbnail(false);
+    expected.setRemoved_original_directory(false);
+    expected.setRemoved_duplicate_file_count(0);
+    expected.setTransformed_name(false);
+    return expected;
+  }
+
+  private static ExpectedFile transformedFedoraFile(FedoraToBagCsv csv, EasyFile ef) {
+    log.trace("EasyFile = {}" , ef);
+    final boolean removeOriginal = csv.getType().startsWith("original") && ef.getPath().startsWith("original/");
+    final String path = removeOriginal
+        ? ef.getPath().replace("original/","")
+        : ef.getPath();
+    final String file = replaceForbidden(path.replaceAll(".*/",""), forbiddenInFileName);
+    final String folder = replaceForbidden(path.replaceAll("[^/]*$",""), forbiddenInFolders);
+    final String dvPath = folder + file;
+
+    ExpectedFile expected = new ExpectedFile();
+    expected.setDoi(csv.getDoi());
+    expected.setSha1_checksum(ef.getSha1checksum());
+    expected.setEasy_file_id(ef.getPid());
+    expected.setFs_rdb_path(ef.getPath());
+    expected.setExpected_path(dvPath);
+    expected.setAdded_during_migration(false);
+    expected.setRemoved_thumbnail(path.matches(".*thumbnails/.*_small.(png|jpg|tiff)"));
```

Comment:
should we account for uppercase?

Reply:
i thought to have expected_path only filled for files that we expect in the dataverse dataset. I guess that is not possible when the expected_path is part of the primary key. so this implementation means we need to filter on `removed_thumbnail = false`